inst/staticexports/purrr.R

```diff
@@ -121,3 +121,25 @@ imap_dbl <- function(.x, .f, ...) {
 imap_chr <- function(.x, .f, ...) {
   map2_chr(.x, vec_index(.x), .f, ...)
 }
+
+keep <- function(.x, .f, ...) {
+  if (is.logical(.f)) {
+    stopifnot(length(.f) == length(.x))
+    .x[.f]
+  } else {
+    .x[map_lgl(.x, .f, ...)]
+  }
+}
```

Comment:
I'd prefer for these functions to be stricter about their inputs -- instead allowing `.f` to be a string or logical vector or function, it should only be a function. This implementation should work:

```suggestion
keep <- function(.x, .f, ...) {
  .x[vapply(.x, .f, ..., FUN.VALUE = NA)]
}
```

And same for `discard`.

Reply:
That is fine with me. `map_lgl()` interface accepted several so I went with the same idea. 
But we could remove the dependency to `map_lgl` too.  Simple and efficient.